### PR TITLE
refactor(web): banner integration with the new gesture engine 🐵

### PR DESCRIPTION
**Pull request by jahorton**
While #9657 started the process for integrating the new gesture engine with the main keyboard within KMW, it did nothing in regard to the predictive-text banner.  This PR remedies the discrepancy, replacing its input-event interpretation engine with the gesture-recognizer as well.

Having both `VisualKeyboard` and `SuggestionBanner` use the same engine means we can now go through the code and remove all references to the _old_ event-interpretation engines.  (Plural - because I tried to make a generalized one before that never actually got used in both places... just one.)  Of course, the new gesture engine is significantly more 'featured' than the old engines, so we'll likely still get a net filesize increase overall... but at least discarding the old engines should make the increase less severe.

@keymanapp-test-bot skip

### CODE REVIEW

File: web/src/engine/osk/src/visualKeyboard.ts [hunk 1]
```diff
@@ -739,9 +732,11 @@ export default class VisualKeyboard extends EventEmitter<EventMap> implements Ke
     let kbdAspectRatio = layerGroup.offsetWidth / this.kbdDiv.offsetHeight;
     let baseKeyProbabilities = this.kbdLayout.getLayer(this.layerId).getTouchProbabilities(touchKbdPos, kbdAspectRatio);
 
-    if (!keySpec || !this.subkeyGesture || !this.subkeyGesture.baseKey.key) {
+    if (!keySpec /*|| !this.subkeyGesture || !this.subkeyGesture.baseKey.key*/) {
       return baseKeyProbabilities;
-    } else {
+    }
+    // TODO:  predictive-text will need mild rework for integration with new gesture integration patterns.
```
Comment (jahorton) [inline web/src/engine/osk/src/visualKeyboard.ts:738]: One of the noted TODOs.

To remove the old `PendingGesture` and `RealizedGesture` classes, which were referencing a now-defunct type connected to the old engines, I needed to comment out this code.  It wasn't being triggered as of #9698, but that in itself is likely a bit of an oversight.

File: web/src/engine/osk/src/visualKeyboard.ts [hunk 2]
```diff
@@ -1723,8 +1622,10 @@ export default class VisualKeyboard extends EventEmitter<EventMap> implements Ke
       return;
     }
 
-    let sk = this.subkeyGesture;
-    let popup = (sk && sk.isVisible());
+    // // TODO: detect any active gesture
+    // let sk = this.subkeyGesture;
+    // let popup = (sk && sk.isVisible());
```
Comment (jahorton) [inline web/src/engine/osk/src/visualKeyboard.ts:1627]: One of the noted TODOs.

File: web/src/engine/osk/src/visualKeyboard.ts [hunk 3]
```diff
@@ -1773,12 +1674,13 @@ export default class VisualKeyboard extends EventEmitter<EventMap> implements Ke
...
 
     this.keytip?.show(null, false, this);
-    this.subkeyGesture?.clear();
-    this.pendingMultiTap?.cancel();
-    this.pendingSubkey?.cancel();
+
+    // // TODO: Have general collection of gesture objects & ability to clear them as needed.
+    // this.subkeyGesture?.clear();
+    // this.pendingMultiTap?.cancel();
+    // this.pendingSubkey?.cancel();
```
Comment (jahorton) [inline web/src/engine/osk/src/visualKeyboard.ts:1683]: One more noted TODO.

File: web/src/engine/osk/src/input/mouseDragOperation.ts [hunk 4]
```diff
@@ -1,7 +1,56 @@
-import InputEventCoordinate from './inputEventCoordinate.js';
-
 type MouseHandler = (this: GlobalEventHandlers, ev: MouseEvent) => any;
 
+/**
+ * Represents the current location of the current cursor / touchpoint during
+ * an ongoing contact-point event series.  This class standardizes to .pageX
+ * (document) coordinates, rather than .clientX (viewport) coordinates.
+ */
+class InputEventCoordinate {
```
Comment (jahorton) [inline web/src/engine/osk/src/input/mouseDragOperation.ts:8]: This comes straight out of the previous file in GitHub's changed-file listings.  I've simplified it to include only the functions that are still utilized.

File: web/src/engine/osk/src/visualKeyboard.ts [hunk 5]
```diff
@@ -832,79 +827,6 @@ export default class VisualKeyboard extends EventEmitter<EventMap> implements Ke
...
-   *
-   * This facilitates modeling of conventional cancellation gestures where a user would
-   * drag the mouse or touch point off the OSK, as mouse and touch event handlers receive
-   * no input beyond screen boundaries.
-   *
-   * @param baseBounds The baseline interactive bounding area to be adjusted
-   * @param startCoord The initial coordinate of a currently-ongoing input event
-   * @returns
-   */
-  private applyScreenMarginBoundsThresholding(baseBounds: BoundingRect,
```
Comment (jahorton) [inline web/src/engine/osk/src/visualKeyboard.ts:848]: The inspiration behind the "safe zone padding" gesture-engine feature.  As it lives there now, we no longer need it here.

File: web/src/engine/osk/src/input/mouseDragOperation.ts (hunk 4 above)
Comment (jahorton) [reply web/src/engine/osk/src/input/mouseDragOperation.ts:8]: The remaining bits are 'small enough' that it's probably not worth the effort at the immediate moment to shift paradigms for this file's exposed class.

**Comment by jahorton** [comment]
... and it was something really dumb on my end.  There's a big difference between `String.replace` and `String.replaceAll` -- some of the stuff I expected to be treeshaken wasn't getting marked as `@__PURE__` because only the first `@class` annotation in a file was being replaced, rather than all of them.

**Comment by jahorton** [comment]
Pending TODO:  rebase upon #9779 & resolve merge conflicts.

Certain code currently commented-out here may be 100% removed afterward, as #9779 sees said code's reimplementation / re-integration.  (Note some of the new in-comment TODOs in the changelog; they'll be resolved.)

File: web/src/engine/osk/src/banner/banner.ts [hunk 6]
```diff
@@ -302,10 +323,10 @@ export class SuggestionBanner extends Banner {
 
     this.buildInternals(false);
 
-    this.manager = new SuggestionInputManager(this.getDiv());
-    this.events = this.manager.events;
+    // this.manager = new SuggestionInputManager(this.getDiv());
```
Comment (mcdurdin) [inline web/src/engine/osk/src/banner/banner.ts:326]: Delete?

File: web/src/engine/osk/src/banner/banner.ts [hunk 7]
```diff
@@ -287,7 +306,9 @@ export class SuggestionBanner extends Banner {
   private options : BannerSuggestion[] = [];
   private hostDevice: DeviceSpec;
 
-  private manager: SuggestionInputManager;
+  private gestureEngine: GestureRecognizer<BannerSuggestion>;
+
+  // private manager: SuggestionInputManager;
```
Comment (mcdurdin) [inline web/src/engine/osk/src/banner/banner.ts:311]: Delete?

File: web/src/engine/osk/src/banner/banner.ts [hunk 8]
```diff
@@ -342,34 +363,119 @@ export class SuggestionBanner extends Banner {
...
+    const config: GestureRecognizerConfiguration<BannerSuggestion> = {
+      targetRoot: this.getDiv(),
+      maxRoamingBounds: new PaddedZoneSource(this.getDiv(), [-0.333 * this.height]),
+      // touchEventRoot:  this.element, // is the default
+      itemIdentifier: (sample, target: HTMLElement) => {
+        let bestMatch: BannerSuggestion = null;
+        let bestDist = Number.MAX_VALUE;
+
+        for(let i = 0; i < this.options.length; i++) {
+          const option = this.options[i];
```
Comment (mcdurdin) [inline web/src/engine/osk/src/banner/banner.ts:391]:
```suggestion
        for(const option of this.options) {
```

Seems to be an option?

File: web/src/engine/osk/src/banner/banner.ts [hunk 9]
```diff
@@ -342,34 +363,119 @@ export class SuggestionBanner extends Banner {
...
+          const suggestion = sample.item;
+
+          if(suggestion != sourceTracker.suggestion) {
+            sourceTracker.suggestion.highlight(false);
+            suggestion.highlight(true);
+            sourceTracker.suggestion = suggestion;
+          }
+        };
+      sourceTracker.suggestion = source.currentSample.item
+
```
Comment (mcdurdin) [inline web/src/engine/osk/src/banner/banner.ts:441]:
```suggestion
      sourceTracker.suggestion = source.currentSample.item;
```

File: web/src/engine/osk/src/input/gestures/browser/pendingMultiTap.ts [hunk 10]
```diff
@@ -1,15 +1,15 @@
...
+// import PendingGesture from '../pendingGesture.interface.js';
 
 export enum PendingMultiTapState { Waiting, Realized, Cancelled };
 /**
  * Implements the multi-tap gesture, which is a series of taps on a single key
  * (based on key id substring in the case of the shift key), within a
  * specified timeout period.
  */
-export default class PendingMultiTap implements PendingGesture {
+export default class PendingMultiTap /*implements PendingGesture*/ {
```
Comment (mcdurdin) [inline web/src/engine/osk/src/input/gestures/browser/pendingMultiTap.ts:12]: why commented out?

File: web/src/engine/osk/src/visualKeyboard.ts [hunk 11]
```diff
@@ -1571,88 +1461,78 @@ export default class VisualKeyboard extends EventEmitter<EventMap> implements Ke
...
-      this.pendingMultiTap.cancel();
-      this.pendingMultiTap = null;
-    }
+  // /**
+  //  * Initializes all supported gestures given a base key and the triggering touch coordinates.
+  //  * @param key     The gesture's base key
+  //  * @param touch   The starting touch coordinates for the gesture
+  //  * @returns
+  //  */
+  // initGestures(key: KeyElement, input: InputEventCoordinate) {
```
Comment (mcdurdin) [inline web/src/engine/osk/src/visualKeyboard.ts:1470]: Commented code to delete

**Review by mcdurdin — APPROVED** [review]

File: web/src/engine/osk/src/input/gestures/browser/pendingMultiTap.ts (hunk 10 above)
Comment (jahorton) [reply web/src/engine/osk/src/input/gestures/browser/pendingMultiTap.ts:12]: `PendingGesture` has been removed from the code base.  Just... forgot to go back and polish this ol' reference up.

**Pull request merged**